cmd/resource-topology-exporter/main.go

```diff
@@ -41,176 +51,103 @@ func main() {
+
+	pArgs.version = flags.Bool("version", false, "Output version and exit")
+
+	err = flags.Parse(args)
 	if err != nil {
-		return nrtupdaterArgs, resourcemonitorArgs, rteArgs, fmt.Errorf("invalid --sleep-interval specified: %w", err)
-	}
-	if ns, ok := arguments["--watch-namespace"].(string); ok {
-		resourcemonitorArgs.Namespace = ns
-	}
-	if kubeletConfigPath, ok := arguments["--kubelet-config-file"].(string); ok {
-		resourcemonitorArgs.KubeletConfigFile = kubeletConfigPath
-	}
-	resourcemonitorArgs.SysfsRoot = arguments["--sysfs"].(string)
-	if path, ok := arguments["--podresources-socket"].(string); ok {
-		resourcemonitorArgs.PodResourceSocketPath = path
+		return pArgs, err
 	}
 
-	if kubeletStateDirs, ok := arguments["--kubelet-state-dir"].([]string); ok {
-		resourcemonitorArgs.KubeletStateDirs = kubeletStateDirs
+	if reflect.DeepEqual(*pArgs.rte.ReferenceContainer, podrescli.ContainerIdent{}) {
+		pArgs.rte.ReferenceContainer = podrescli.ContainerIdentFromEnv()
 	}
 
-	rteArgs.Debug = arguments["--debug"].(bool)
-	if refCnt, ok := arguments["--reference-container"].(string); ok {
-		rteArgs.ReferenceContainer, err = resourcetopologyexporter.ContainerIdentFromString(refCnt)
-		if err != nil {
-			return nrtupdaterArgs, resourcemonitorArgs, rteArgs, err
-		}
-	}
-	if rteArgs.ReferenceContainer == nil {
-		rteArgs.ReferenceContainer = resourcetopologyexporter.ContainerIdentFromEnv()
-	}
+	return pArgs, nil
+}
 
-	if configPath, ok := arguments["--config"].(string); ok {
-		conf, err := readConfig(configPath)
+func defaultHostName() string {
+	var err error
+
+	val, ok := os.LookupEnv("NODE_NAME")
+	if !ok || val == "" {
+		val, err = os.Hostname()
 		if err != nil {
-			log.Fatalf("error getting exclude list from the configuration: %v", err)
+			log.Fatalf("error getting the host name: %v", err)
 		}
-		resourcemonitorArgs.ExcludeList.ExcludeList = conf.ExcludeList
-		log.Printf("using exclude list:\n%s", resourcemonitorArgs.ExcludeList.String())
 	}
-	if tmPolicy, ok := arguments["--topology-manager-policy"].(string); ok {
-		if tmPolicy == "" {
-			// last attempt
-			tmPolicy = os.Getenv("TOPOLOGY_MANAGER_POLICY")
-		}
-		// empty string is a valid value here, so just keep going
-		rteArgs.TopologyManagerPolicy = tmPolicy
-	}
-	return nrtupdaterArgs, resourcemonitorArgs, rteArgs, nil
+	return val
 }
 
-type config struct {
-	ExcludeList map[string][]string
-}
-
-func readConfig(configPath string) (config, error) {
-	conf := config{}
-	data, err := os.ReadFile(configPath)
-	if err != nil {
-		// config is optional
-		if errors.Is(err, os.ErrNotExist) {
-			log.Printf("Info: couldn't find configuration in %q", configPath)
-			return conf, nil
-		}
-		return conf, err
+func defaultTmPolicy() string {
```

Comment:
I think here we can afford to write `defaultTopologyManagerPolicy`

Reply:
Sure